### PR TITLE
feat(ecocredit/core): SealBatch server impl

### PR DESCRIPTION
**Pull request by technicallyty**
## Description

- implements the SealBatch server method
- adds seal event

ref: #973 

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: x/ecocredit/server/core/seal_batch.go [hunk 1]
```diff
@@ -3,9 +3,41 @@ package core
...
+		return nil, err
+	}
+
+	batch, err := k.stateStore.BatchTable().GetByDenom(ctx, req.BatchDenom)
+	if err != nil {
+		return nil, sdkerrors.ErrInvalidRequest.Wrapf("could not get batch with denom %s: %s", req.BatchDenom, err.Error())
+	}
+
+	if !batch.Open {
+		return &core.MsgSealBatchResponse{}, nil
```
Comment (aleem1314) [inline x/ecocredit/server/core/seal_batch.go:26]: Can we add a test for this?

Comment (aleem1314) [inline x/ecocredit/server/core/seal_batch.go:26]: Can we add a test for this?

Comment (technicallyty) [reply x/ecocredit/server/core/seal_batch.go:26]: https://github.com/regen-network/regen-ledger/pull/1060/commits/578cb0e18e0313802d4aad48f09b54e502f0c21a

Comment (technicallyty) [reply x/ecocredit/server/core/seal_batch.go:26]: i was unsure which direction to go - we do a no-op in x/data when an attestor attests to data they've already attested to so i had that in mind when going this route. 

would appreciate more opinions here CC: @ryanchristo @aaronc

**Review by ryanchristo — COMMENTED** [review]
Looking good. Just a couple comments.

**Review by ryanchristo — APPROVED** [review]
Looks good to me.

**Review by aleem1314 — APPROVED** [review]
lgtm

**Comment by ryanchristo** [comment]
@technicallyty can you fix conflicts? :pray:

**Pull request merged**